webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTableSchema.js

```diff
@@ -0,0 +1,65 @@
+import URI from 'urijs';
+import {
+  ellipsisCellFormatter,
+  cellFormatter,
+  headerFormatterWithProps,
+  cellFormatterWithProps,
+  nameCellFormatter,
+  hostsCountCellFormatter,
+  deleteActionCellFormatter,
+  sortableHeaderFormatter,
+} from '../common/table';
+
+const sortableColumn = (
+  property,
+  label,
+  mdWidth,
+  sortController,
+  additionalCellFmt = [],
+) =>
+  column(
+    property,
+    label,
+    [sortableHeaderFormatter(sortController), headerFormatterWithProps],
+    [...additionalCellFmt, ellipsisCellFormatter],
+    { sort: true, sortDirection: '', className: `col-md-${mdWidth}` },
+  );
+
+const column = (property, label, headFormat, cellFormat, headProps = {}, cellProps = {}) => ({
+  property,
+  header: {
+    label: __(label),
+    props: headProps,
+    formatters: headFormat,
+  },
+  cell: {
+    props: cellProps,
+    formatters: cellFormat,
+  },
+});
+
+export const sortControllerFactory = (apiCall, sortBy, sortOrder) => ({
+  apply: (by, order) => {
+    const uri = new URI(window.location.href);
+    uri.setSearch('order', `${by} ${order}`);
+    // FIXME(bshuster): Going back won't render the state.
+    //                  Using react-router will fix this completely.
+    window.history.pushState({ path: uri.toString() }, '', uri.toString());
+    apiCall('models', uri.query(true));
+  },
+  property: sortBy,
+  order: sortOrder,
+});
+
```

Comment:
I have a feeling that everything above this line is generic and less relevant to the models page? I was just trying to understand when someone wants to reproduce another table, which part he actually is going to copy/paste?